### ISSUE TITLE
account/user limits allows for "undefined" datacenter

### ISSUE THREAD

**Issue by stensonb** [joyent/sdc-adminui#208]
I'm seeing a "undefined" table entry on the provisioning limits page...which seems...inelegant.

<img width="1063" alt="screen shot 2016-06-10 at 11 02 26 am" src="https://cloud.githubusercontent.com/assets/933389/15973980/f3417b66-2efa-11e6-8815-e6e36621d071.png">

I can observe the UI is making a call to the cloudapi, and it gets a response like this:
```json
[
  {
    "dn": "dclimit=tacoma, uuid=0efe93a4-a0dd-49af-c127-c9df46352694, ou=users, o=smartdc",
    "datacenter": "tacoma",
    "limit": [
      {
        "check": "os",
        "os": "any",
        "by": "ram",
        "value": "2000"
      }
    ]
  },
  {
    "dn": "dclimit=undefined, uuid=0efe93a4-a0dd-49af-c127-c9df46352694, ou=users, o=smartdc",
    "datacenter": "undefined",
    "limit": []
  }
]
```

So, the UI is displaying each element of the array, but the data in the array is bad.

Here's the problem.  This UI allowed me to create a bad array element by creating a limit WITHOUT specifying a datacenter.  And, further, if I update/modify a new limit WITHOUT specifying the dc, the "undefined" array entry gets update with my values - though the record is still titled "undefined".  If I then delete the limit from the "undefined" dc, that limit is removed from array, but the "undefined" dc still exists in the array.

I'm not sure what the intent is here.  If the datacenter is a requirement for setting a limit, the UI should prevent a POST to the API without the missing data.  Furthermore, the API should validate required data exists and prevent any update if the validation fails.

Alternatively, if the datacenter is NOT a requirement, and instead this is an attempt to allow for a "global" or "all datacenters" limits to be managed, then the "undefined" label should be changed to something more user-friendly.

**Comment by askfongjojo** [comment]
Hi @stensonb - I am sorry for the delay in response. The issue has been filed as https://smartos.org/bugview/ADMINUI-2317 to tighten up the input validation for the datacenter field. We'll have a fix for this soon.

**Comment by askfongjojo** [comment]
@stensonb - The issue has been fixed with c3ff5e5 (where a blank DC is no longer allowed).

To clean up the 'undefined' DC limit record, please use the sdc-ldap CLI:
```
sdc-ldap delete 'dclimit=undefined, uuid=${account-uuid}, ou=users, o=smartdc'
```
